### ISSUE TITLE
IFC4 Schema - Import File

### ISSUE THREAD

**Issue by SeppArc** [BIM-Tools/SketchUp-IFC-Manager#64]
Hi,

when importing an Ifc file with Scheme IFC4 and also with the settings of the IFC Manager set to IFC4, only the tag is being populated but not the necessary IfcType:

![image](https://github.com/user-attachments/assets/cfd0f326-fead-456a-a4f9-4749a73b8326)

If changing the settings to IFC2X3 and doing the Import again, the IfcType is being populated. Switching back to IFC4, the type will be lost again.

Please help me, my customers need IFC4.

**Comment by jbrouwerdigibase** [comment]
The IFC import button in IFC manager just calls the native Sketchup importer and does some cleanup. I don't use it much myself and never noticed that it imports IFC4 and IFC4x3 but 'downgrades' them to IFC2x3.
I just tested with an IFC4x3 model that has an IfcRoad and IfcBridge, and all these new entities end up as IfcBuildingElementProxy.
The best workable solution is something of a function that copies IFC data to another version, would that help? It would not fix the "downgrading" but would allow you to work on the imported model in the IFC version you want.

**Comment by claumeh** [comment]
Yes, your proposed solution would be sufficient for most of my usecases. If after the importing of an IFC4 model, IfcWalls, Slabs, Columns are represented in IFC4 and I can work with it that would help a lot.
Thanks!
